Marlin/ultralcd.cpp

```diff
@@ -4762,6 +4777,13 @@ void lcd_update() {
 
     #if ENABLED(ULTIPANEL)
 
+      // if SD_REPRINT_LAST_SELECTED_FILE is active, we suppress time outs from the
+      // void lcd_sdcard_menu() routine.  The LCD Panel should not be timing out after
+      // a print when waiting for the user to select the same print again.
+      #if ENABLED(SD_REPRINT_LAST_SELECTED_FILE)
+        if (currentScreen != lcd_sdcard_menu)
+      #endif
+  
```

Comment:
This is what `defer_return_to_status` is for. Just set it in `lcd_sdcard_menu`. It is cleared automatically after the user click.

Reply:
No...  There is another issue.    Without the explicit check for the current screen being lcd_sdcard_menu...    Some times at the end of the print the status screen keeps being shown even though the   lcd_goto_screen(lcd_sdcard_menu, saved_encoderPosition);  has been called from inside void CardReader::printingHasFinished()

It seems there is a race condition in the LCD code, and depending on when the lcd_goto_screen() is called, some times it won't update the screen.   BUT...  it is clearly in the correct screen.  It just doesn't show the correct screen until something like an encoder tick happens.

Reply:
Rather than perpetuating the issue with hacks and workarounds, it would be more sensible to deal with that issue first.

Reply:
I'll see if I can find the root cause issue and repair it.   But LCD code is mess.  It needs to be rewritten.